### PR TITLE
Fix 787: Add support for custom payment gateway icons

### PR DESCRIPTION
**Pull request by jaapmarcus**
Add custom logo's for any provider

### CODE REVIEW

**Comment by jaapmarcus** [comment]
Before: https://github.com/FOSSBilling/FOSSBilling/issues/787
After:
Screenshot 2023-02-25 at 23 39 39

Example:
https://github.com/jaapmarcus/mollie-fossbilling/blob/main/Mollie/Mollie.php

**Comment by jaapmarcus** [comment]
Will break existing custom themes as then need to add the code for the css generation them self.

To allow the user to create the css them self I have decide to supply the height, width and logo separate instead of the css code

**Comment by jaapmarcus** [comment]
Screenshot 2023-02-26 at 02 19 33

**Comment by BelleNottelling** [comment]
> Will break existing custom themes as then need to add the code for the css generation them self.

Are you sure about this?
1. Unless they specifically override the file in their theme, the new method will be used
2. If they do have an outdated theme that hasn't been updated for this, wouldn't it just end up using the old method with the CSS classes?

**Comment by jaapmarcus** [comment]
> 1. Unless they specifically override the file in their theme, the new method will be used
> 2. If they do have an outdated theme that hasn't been updated for this, wouldn't it just end up using the old method with the CSS classes?

True..  Unless the load the css / logos in from Hugaru theme.. 

Anyway it will be more flexible and future proof the code...

**Comment by BelleNottelling** [comment]
Also, it seems we have two upload directories. `/uploads` and `/data/uploads`. We should decide on one and remove the other.
My personal vote is for `/data/uploads`

**Comment by jaapmarcus** [comment]
> Also, it seems we have two upload directories. `/uploads` and `/data/uploads`. We should decide on one and remove the other.
> My personal vote is to keep `/data/uploads`

I don't mind either of them

Ideally we should create 2 folders:
1. For the logo assets  from gateways and so an 
2. For the uploads

File: src/modules/Invoice/ServicePayGateway.php [hunk 1]
```diff
@@ -209,10 +209,25 @@ public function getActive(array $data)
...
             } else {
-                $result[] = $this->toApiArray($gtw);
+                $gateway =  $this->toApiArray($gtw);
+                $adapter = $this -> getPaymentAdapter($gtw);
+                if ( array_key_exists('logo',$adapter->getConfig())){
+                    $gateway['logo'] = $adapter->getConfig()['logo'];
+                    if(file_exists(PATH_LIBRARY . '/Payment/Adapter/'.$adapter->getConfig()['logo']['logo'])){
+                        $gateway['logo']['logo'] = $this->di['tools']->url('/library/Payment/Adapter/' . $adapter->getConfig()['logo']['logo']);
+                    }else{
+                        if(file_exists(PATH_ROOT . '/uploads/gateways/'. $adapter->getConfig()['logo']['logo'])){
```
Comment (BelleNottelling) [inline src/modules/Invoice/ServicePayGateway.php:219]: code still points to the deleted folder

Comment (jaapmarcus) [reply src/modules/Invoice/ServicePayGateway.php]: Changed

**Review by BelleNottelling — APPROVED** [review]

**Review by John-S4 — APPROVED** [review]

**Pull request merged**